src/cdk/a11y/key-manager/tree-key-manager.ts

```diff
@@ -0,0 +1,559 @@
+ */
+export interface TreeKeyManagerOptions<T extends TreeKeyManagerItem> {
+  /**
+   * Sets the predicate function that determines which items should be skipped by the tree key
+   * manager. By default, disabled items are skipped.
+   *
+   * If the item is to be skipped, this function should return false.
+   */
+  skipPredicate?: (item: T) => boolean;
+
+  /**
+   * If true, then the key manager will call `activate` in addition to calling `focus` when a
+   * particular item is focused. By default, this is false.
+   */
+  activationFollowsFocus?: boolean;
+
+  /**
+   * The direction in which the tree items are laid out horizontally. This influences which key
+   * will be interpreted as expand or collapse. Defaults to 'ltr'.
+   */
+  horizontalOrientation?: 'rtl' | 'ltr';
+
+  /**
+   * If provided, determines how the key manager determines if two items are equivalent.
+   *
+   * It should provide a unique key for each unique tree item. If two tree items are equivalent,
+   * then this function should return the same value.
+   */
+  trackBy?: (treeItem: T) => unknown;
+
+  /**
+   * If a value is provided, enables typeahead mode, which allows users to set the active item
+   * by typing the visible label of the item.
+   *
+   * If a number is provided, this will be the time to wait after the last keystroke before
+   * setting the active item. If `true` is provided, the default interval of 200ms will be used.
+   */
+  typeAheadDebounceInterval?: true | number;
+}
+
+export interface TreeKeyManagerStrategy<T extends TreeKeyManagerItem> {
+  /** Stream that emits any time the focused item changes. */
+  readonly change: Subject<T | null>;
+
+  /**
+   * Handles a keyboard event on the tree.
+   *
+   * @param event Keyboard event that represents the user interaction with the tree.
+   */
+  onKeydown(event: KeyboardEvent): void;
+
+  /** Index of the currently active item. */
+  getActiveItemIndex(): number | null;
+
+  /** The currently active item. */
+  getActiveItem(): T | null;
+
+  /**
+   * Called the first time the Tree component is focused. This method will only be called once over
+   * the lifetime of the Tree component.
+   *
+   * Intended to be used to focus the first item in the tree.
+   */
+  onInitialFocus(): void;
+
+  /**
+   * Focus the provided item by index.
+   *
+   * Updates the state of the currently active item. Emits to `change` stream if active item
+   * Changes.
+   * @param index The index of the item to focus.
+   * @param options Additional focusing options.
+   */
+  focusItem(index: number, options?: {emitChangeEvent?: boolean}): void;
+  /**
+   * Focus the provided item.
+   *
+   * Updates the state of the currently active item. Emits to `change` stream if active item
+   * Changes.
+   * @param item The item to focus. Equality is determined via the trackBy function.
+   * @param options Additional focusing options.
+   */
+  focusItem(item: T, options?: {emitChangeEvent?: boolean}): void;
```

Comment:
This is a good catch, didn't consider if `T` is a `number`.

Currently in main, `T` can be any type. I would want to change it to be `<T extends {}>` because I don't think the Tree intends for T to be primitives. That's not in scope for this PR, but it would be a great thing to test in a hackathon :).